src/OrleansAzureUtils/IServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Orleans;
+using Orleans.AzureUtils.Configuration;
+using Orleans.Runtime.Configuration;
+using Orleans.Runtime.MembershipService;
+
+namespace OrleansAzureUtils
+{
+    public static class IServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Configure DI container to use AzureBasedMemebershipTable
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="configureOptions"></param>
+        public static IServiceCollection UseAzureMemebershipTable(this IServiceCollection services,
+            Action<AzureMembershipTableOptions> configureOptions)
+        {
+            services.Configure<AzureMembershipTableOptions>(configureOptions);
+            services.AddSingleton<IMembershipTable, AzureBasedMembershipTable>();
+            return services;
+        }
+
+        /// <summary>
+        /// Configure DI container to use AzureBasedMemebershipTable
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="config"></param>
+        public static IServiceCollection UseAzureMemebershipTable(this IServiceCollection services, IConfiguration config)
+        {
+            services.Configure<AzureMembershipTableOptions>(config);
+            services.AddSingleton<IMembershipTable, AzureBasedMembershipTable>();
+            return services;
+        }
+
+
+        /// <summary>
+        /// Configure DI to use AzureBasedMemebershipTable, and get its configuration from <see cref="GlobalConfiguration"/>.
+        /// </summary>
+        /// <param name="services"></param>
+        /// <returns></returns>
+        public static IServiceCollection UseAzureMembershipTableFromLegacyConfigurationSupport(
```

Comment:
@jdom, @xiazen: What is the plan with the legacy configuration extension methods long term? We did not think about those for these extensions IIRC. Moving them to .Legacy packages once we moved legacy config classes too?

Reply:
See https://github.com/dotnet/orleans/pull/3490#discussion_r142747242